.github/workflows/build-docs.yaml

```diff
@@ -37,25 +37,11 @@ jobs:
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
         continue-on-error: true
 
-      - name: restore cargo cache
-        uses: actions/cache@v3
-        with:
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock', 'rust-toolchain.toml') }}
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-            .cargo/.package-cache/
-            .foundry/bin/
+      - name: Build bindings
+        run: nix develop -c make build-solidity-types
 
       - name: Generate rust docs
-        uses: workflow/nix-shell-action@v3.3.0
-        with:
-          flakes-from-devshell: true
-          script: |
-            cargo doc --no-deps
+        run: nix develop -c cargo doc --no-deps
```

Comment:
The flakes-from-devshell should be a superior method.

Reply:
How so? It is literally only an additional wrapper around the same command.

Reply:
I don't think that warrants having to care for another GH action. Also non-nix users won't touch this.